jmix-core/core/src/main/java/io/jmix/core/Copier.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.core;
+
+/**
+ * Creates copies of objects.
+ */
+public interface Copier {
+
+    /**
+     * Creates a deep copy of the passed object.
```

Comment:
1. Better to add `@return` and `@param` to avoid warnings from `javaDoc` task.
2. Let's describe in more detail what "deep copy" means. At least, similar to `io.jmix.core.MetadataTools#deepCopy(T)`. Describe the differences from `MetadataTools#deepCopy(T)` as it may be unclear what to use and when.

Reply:
Done